lib/active_model/serializer/adapter/json_api/error.rb

```diff
@@ -0,0 +1,75 @@
+require 'active_model_serializers/json_pointer'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi < Base
+        module Error
+          UnknownSourceTypeError = Class.new(ArgumentError)
+
+          # @param [ActiveModel::Serializer::ErrorSerializer]
+          # @return [Array<Symbol, Array<String>] i.e. attribute_name, [attribute_errors]
+          def self.resource_errors(error_serializer)
+            error_serializer.as_json.flat_map do |attribute_name, attribute_errors|
+              attribute_error_objects(attribute_name, attribute_errors)
+            end
+          end
+
+          # definition:
+          #   JSON Object
+          #
+          # properties:
+          #   ☐ id      : String
+          #   ☐ status  : String
```

Comment:
I have no idea how we'll implement this stuff, but it's here, and we can look at https://github.com/cerebris/jsonapi-resources/blob/8e85d68dfbaf9181344c7618b0b29b4cfd362034/lib/jsonapi/response_document.rb#L22-L28 https://github.com/cerebris/jsonapi-resources/blob/8e85d68dfbaf9181344c7618b0b29b4cfd362034/lib/jsonapi/operation_result.rb#L3-L14